BusBoard.ConsoleApp/Program.cs

```diff
@@ -1,39 +1,57 @@
 ﻿using System;
-using System.Collections.Generic;
-using System.Linq;
-using System.Text;
-using System.Threading.Tasks;
-using Newtonsoft.Json;
-using RestSharp;
-using RestSharp.Authenticators;
 
 namespace BusBoard.ConsoleApp
 {
   class Program
   {
     static void Main(string[] args)
     {
-      Console.WriteLine("Please enter your desired stop point ID:");
-      var stopCode = Console.ReadLine();
-      Console.WriteLine(" ");
-      var buses = GetNext5BusesForStopPoint(stopCode);
-      Console.WriteLine($"Next 5 Buses at stop {stopCode}");
-      foreach (var bus in buses)
+      TflApi api = new TflApi();
+      PromptUserAndGetInputFromConsole(api);
+    }
+
+    private static void PromptUserAndGetInputFromConsole(TflApi api)
+    {
+      Console.WriteLine("Postcode or Stop point ID? (P or S)");
+      switch (Console.ReadLine().ToUpper())
       {
-        Console.WriteLine($"VehicleID: {bus.VehicleId}, ETA: {bus.ExpectedArrival.AddHours(1).ToString("HH:mm:ss tt")} ({bus.TimeToStation} seconds away)");
+        case "S":
+          Console.WriteLine("Please enter your desired stop point ID:");
+          var stopCode = Console.ReadLine();
+          Console.WriteLine(" ");
+          PrintBusesFromStopCode(stopCode, "", api);
+          break;
+        case "P":
+          Console.WriteLine("Please enter your desired postcode");
+          var postcode = Console.ReadLine();
+          Console.WriteLine(" ");
+          foreach (var sc in api.GetTwoClosestBusStopsToPostcode(postcode))
+          {
+            PrintBusesFromStopCode(sc.NaptanId, sc.CommonName, api);
+          }
+
+          break;
+        default:
+          Console.WriteLine("Please enter S or P");
+          break;
       }
-      Console.WriteLine(" ");
     }
 
-    private static IEnumerable<BusArrivalPrediction> GetNext5BusesForStopPoint(string stopCode)
+    private static void PrintBusesFromStopCode(string stopCode, string commonName, TflApi api)
```

Comment:
I think this method is doing too much and we could separate it out a bit. I think it is both getting the predictions and displaying them to the console.

Maybe the call to the API should be brought outside this method and in to the main method, and this method could just have the display logic and takes it some bus predictions

Reply:
Ok, makes sense